org.jacoco.core.test/src/org/jacoco/core/test/filter/targets/Constructor.java

```diff
@@ -0,0 +1,67 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2017 Mountainminds GmbH & Co. KG and Contributors
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Evgeny Mandrikov - initial API and implementation
+ *
+ *******************************************************************************/
+package org.jacoco.core.test.filter.targets;
+
+/**
+ * This test target is a constructors.
+ */
+public class Constructor {
+
+	Constructor() { // $line-packageLocal$
+	}
+
+	private Constructor(Object arg) { // $line-arg$
+	}
+
+	private static class Super extends Constructor {
+		private Super() {
+			super(null); // $line-super$
+		}
+	}
+
+	private class Inner {
+		private Inner() { // $line-inner$
+		}
+	}
+
+	private static class InnerStatic {
+		private Object field = this;
+
+		private InnerStatic() { // $line-innerStatic$
+		}
+	}
+
+	public static class PublicDefault { // $line-publicDefault$
+	}
+
+	static class PackageLocalDefault { // $line-packageLocalDefault$
+	}
+
+	private static class PrivateDefault { // $line-privateDefault$
+	}
+
+	private static class PrivateEmptyNoArg {
```

Comment:
@Godin I wonder whether we should add another negative test for the case where the private non-arg constructor is not empty in source code: 

	private static class PrivateNonEmptyNoArg {
		private PrivateEmptyNoArg() {
			nop(); // $line-privateNonEmptyNoArg$
		}
	}